src/content/en/tools/workbox/guides/using-plugins.md

```diff
@@ -103,54 +103,112 @@ A plugin using all of these callbacks would look like this:
+  cacheKeyWillBeUsed: async ({request, mode, params, event, state}) => {
     // `request` is the `Request` object that would otherwise be used as the cache key.
     // `mode` is either 'read' or 'write'.
     // Return either a string, or a `Request` whose `url` property will be used as the cache key.
     // Returning the original `request` will make this a no-op.
     return request;
   },
-  cachedResponseWillBeUsed: async ({cacheName, request, matchOptions, cachedResponse, event}) => {
+  cachedResponseWillBeUsed: async ({cacheName, request, matchOptions, cachedResponse, event, state}) => {
     // Return `cachedResponse`, a different `Response` object, or null.
     return cachedResponse;
   },
-  requestWillFetch: async ({request}) => {
+  requestWillFetch: async ({request, event, state}) => {
     // Return `request` or a different `Request` object.
     return request;
   },
-  fetchDidFail: async ({originalRequest, request, error, event}) => {
+  fetchDidFail: async ({originalRequest, request, error, event, state}) => {
     // No return expected.
     // NOTE: `originalRequest` is the browser's request, `request` is the
     // request after being passed through plugins with
     // `requestWillFetch` callbacks, and `error` is the exception that caused
     // the underlying `fetch()` to fail.
   },
-  fetchDidSucceed: async ({request, response}) => {
+  fetchDidSucceed: async ({request, response, event, state}) => {
     // Return `response` to use the network response as-is,
     // or alternatively create and return a new `Response` object.
     return response;
   }
 };
 ```
 
-Note: the `event` object passed to each plugin callback above represents the
-original event that triggered the fetch or cache action. In some cases there
-will **not** be an original event, so your code should check for its existence
-before referencing it. Also, when invoking the
-[`handle()`](/web/tools/workbox/guides/advanced-recipes#handle)
-method of a strategy, the `event` you pass to `handle()` will be the event
-passed to the plugin callbacks.
+The `event` object passed to each plugin callback above represents the original event that triggered
+the fetch or cache action. In some cases there will **not** be an original event, so your code
+should check for its existence before referencing it. Also, when invoking the
+[`handle()`](/web/tools/workbox/guides/advanced-recipes#handle) method of a strategy, the `event`
+you pass to `handle()` will be the event passed to the plugin callbacks.
+
+All plugin callbacks are also passed a `state` object which is unique to the particular plugin
+object and strategy invocation (i.e. the call to `handle()`). This makes it possible to write
+plugins where one callback can conditionally do something based on what another callback in the same
+plugin did (e.g. compute the time delta between running `requestWillFetch()` and `fetchDidSucceed()`
+or `fetchDidFail()`).
+
+## Lifecycle Callbacks
```

Comment:
I don't think there's a meaningful distinction between these "lifecycle callbacks" and the other group of existing callbacks. All of the callbacks in this section are new for v6 and are tied to the `handler*` lifecycle, but conceptually, they're very similar to the existing callbacks and should probably be listed alongside them.

Would you mind rearranging things so that these aren't broken out into their own section? You can add in something like "The lifecycle callbacks starting with `handler` are all new in Workbox v6." to make it clear what the support story is.

Reply:
Updated, and added a note mentioning how `handler` callbacks are new in v6 :)